### PR TITLE
fix: replace replaceAll() with regex

### PR DESCRIPTION
**Pull request by anantjain45823**
## Description of the change

> Replacing the use of replaceAll() in #32 with regex expression as replaceAll() is not supported in currently used node v14. 

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: router.js [hunk 1]
```diff
@@ -45,9 +45,9 @@ router.get("/load", async (ctx) => {
   d = d.replace("dataPlaneUrl", dataPlaneUrl);
   d = d.replace("configBackendUrl", configUrl);
 
-  deviceModeInit = deviceModeInit.replaceAll("dataplaneUrl", dataPlaneUrl);
-  deviceModeInit = deviceModeInit.replaceAll("writeKey_placeHolder", writeKey);
-  deviceModeInit = deviceModeInit.replaceAll("configUrl_placeholder", configUrl);
+  deviceModeInit = deviceModeInit.replace(/dataplaneUrl/g, dataPlaneUrl);
```
Comment (sanpj2292) [inline router.js:48]: Do we have test-case for this ? I mean not literally for `replace(/dataplaneUrl/g, "")` but more like expressing the intention of this kind of replace implementation.

File: router.js [hunk 2]
```diff
@@ -45,9 +45,9 @@ router.get("/load", async (ctx) => {
...
+  deviceModeInit = deviceModeInit.replace(/writeKey_placeHolder/g, writeKey);
+  deviceModeInit = deviceModeInit.replace(/writeKey_placeHolder/g, configUrl);
```
Comment (utsabc) [inline router.js:50]: Why are we overriding. `writeKey_placeHolder` with configUrl?

Comment (sanpj2292) [reply router.js:50]: Seems like a copy-paste thing. Let's get this fixed ?

File: router.js (hunk 1 above)
Comment (anantjain45823) [reply router.js:48]: Didn't understand your point. Will changing it to `dataplaneUrl_placeHolder` would make sense?

File: router.js (hunk 2 above)
Comment (anantjain45823) [reply router.js]: Done

File: router.js [hunk 3]
```diff
@@ -45,9 +45,9 @@ router.get("/load", async (ctx) => {
   d = d.replace("dataPlaneUrl", dataPlaneUrl);
   d = d.replace("configBackendUrl", configUrl);
 
-  deviceModeInit = deviceModeInit.replaceAll("dataplaneUrl", dataPlaneUrl);
-  deviceModeInit = deviceModeInit.replaceAll("writeKey_placeHolder", writeKey);
-  deviceModeInit = deviceModeInit.replaceAll("configUrl_placeholder", configUrl);
+  deviceModeInit = deviceModeInit.replace(/dataplaneUrl_placeHolder/g, dataPlaneUrl);
+  deviceModeInit = deviceModeInit.replace(/writeKey_placeHolder/g, writeKey);
+  deviceModeInit = deviceModeInit.replace(/configUrl_placeholder/g, configUrl);
```
Comment (utsabc) [inline router.js:50]: I think in device mode init the config url is stored in a variable called webhookURL can we change that

Comment (anantjain45823) [reply router.js:50]: Why do we need to change that it is used to fetch the` /sourceConfig`

**Review by yashasvibajpai — APPROVED** [review]

**Pull request merged**